rosbag2_cpp/include/rosbag2_cpp/writer.hpp

```diff
@@ -199,6 +199,26 @@ class ROSBAG2_CPP_PUBLIC Writer
     const std::string & type_name,
     const rclcpp::Time & time);
 
+  /**
+   * Write a serialized message to a bagfile.
+   * The topic will be created if it has not been created already.
+   *
+   * \warning after calling this function, the serialized data will no longer be managed by message.
+   *
+   * \param message rclcpp::SerializedMessage The serialized message to be written to the bagfile
+   * \param topic_name the string of the topic this messages belongs to
+   * \param type_name the string of the type associated with this message
+   * \param recv_time The time stamp when the message was received
+   * \param send_time The time stamp when the message was send
+   * \throws runtime_error if the Writer is not open.
+   */
+  void write(
+    std::shared_ptr<const rclcpp::SerializedMessage> message,
+    const std::string & topic_name,
+    const std::string & type_name,
+    const rclcpp::Time & recv_time,
+    const rclcpp::Time & send_time);
+
```

Comment:
```suggestion
  void write(
    std::shared_ptr<const rclcpp::SerializedMessage> message,
    const std::string & topic_name,
    const std::string & type_name,
    const rcutils_time_point_value_t & recv_timestamp,
    const rcutils_time_point_value_t & send_timestamp);
```
It doesn't make sense to use `rclcpp::Time` for the newly created API since we will have to convert timestamp twice. It will be more efficient to use `rcutils_time_point_value_t` the same type as in the destination structure. Please note that the source timestamp from message info is also derived from rcutils time point
```
typedef rcutils_time_point_value_t rmw_time_point_value_t;
```